### PR TITLE
docs: dumi

### PR DESCRIPTION
**Pull request by thonatos**
##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

- docs
- ci workflows

##### Description of change

- use dumi to build docs site
- use puppeteer for doc tests
- use github action to deploy gh-pages
- use vercel to build docs site

**Links**

- https://egg-pi.vercel.app/
- https://d.umijs.org/zh-CN/config/frontmatter

### CODE REVIEW

File: docs/.dumi/theme/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React, { useContext } from 'react';
```
Comment (fengmk2) [inline docs/.dumi/theme/layout.tsx:1]: 这些配置为什么放 .dumi 这种隐藏目录？

**Review by fengmk2 — APPROVED** [review]

File: docs/.dumi/theme/layout.tsx (hunk 1 above)
Comment (thonatos) [reply docs/.dumi/theme/layout.tsx:1]: 这是 dumi 的 local theme 目录

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Setup Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Install Dependencies
+      run: npm i -g npminstall && npminstall
+
+    - name: Build Documents
+      run: npm run docs:build
```
Comment (fengmk2) [inline .github/workflows/gh-pages.yml:29]: @thonatos docs:build 这个命令不存在

File: package.json [hunk 3]
```diff
@@ -57,15 +57,18 @@
     "ylru": "^1.2.1"
   },
   "devDependencies": {
+    "@umijs/preset-react": "^1.x",
     "address": "^1.0.3",
+    "antd": "^4.18.6",
```
Comment (fengmk2) [inline package.json:62]: @thonatos antd 是在那里被依赖的？

**Pull request merged**